### ISSUE TITLE
Training stuck generating rollouts

### ISSUE THREAD

**Issue by javirandor** [CarperAI/trlx#485]
### 🐛 Describe the bug

This is a follow-up from issue #399.

I am facing this same issue even with the updated code.

I am trying to reproduce the HH fine-tuning example on Alpaca.

```
trlx.train(
    prompts=prompts,
    eval_prompts=eval_prompts,
    reward_fn=reward_fn,
    config=config,
    stop_sequences=["Human:", "human:", "Assistant:", "assistant:"]
)
```

My code gets stuck generating the second batch of rollouts `16/64`.

I am running the code on a cluster with 8xA100s (80GB) and using a custom reward model. Providing a minimal reproducible example is a bit hard for my current setup. Do you have any pointers that can help me debug this issue?

My accelerate config as taken from the repo

```
compute_environment: LOCAL_MACHINE
deepspeed_config:
  deepspeed_multinode_launcher: standard
  gradient_accumulation_steps: 1
  gradient_clipping: 1.0
  offload_optimizer_device: none
  offload_param_device: none
  zero3_init_flag: false
  zero_stage: 2
distributed_type: DEEPSPEED
downcast_bf16: no
dynamo_config: {}
fsdp_config: {}
machine_rank: 0
main_training_function: main
megatron_lm_config: {}
mixed_precision: bf16
num_machines: 1
num_processes: 8
rdzv_backend: static
same_network: true
use_cpu: false
```

### Which trlX version are you using?

Installed from sourced @355c974

### Additional system and package information

Python 3.9, transformers==4.28.1, torch==2.0.0, accelerate==0.18.0, deepspeed==0.9.1

**Comment by javirandor** [comment]
The code gets stuck in the `gather_dict` function. More specifically, in the line `torch.distributed.all_gather_object(objs, obj)`.

**Comment by javirandor** [comment]
I could solve the problem by commenting out the calls to the `gather_dict` function since they were creating metadata that is not useful in my use case. I would be curious to see if there is a cleaner solution to this issue. It seems to be a [native torch thing](https://github.com/pytorch/pytorch/issues/18689) but the solution suggested there did not solve the problem.

**Comment by maxreciprocate** [comment]
That's a peculiar issue, does it these errors also occur when using example scripts and/or example reward models and base models? And if could you share your launching script, that might be also helpful.

**Comment by maxreciprocate** [comment]
Was not able to reproduce the error, but please do retry with most recent code if the issue still relevant.
